lib/main.dart

```diff
@@ -1,7 +1,50 @@
-void main() {
-  /**
-   *    Welcome to Basic authentication challenge
-   *    Make sure to read CHALLENGE.md before you start
-   *
-   */
+import 'dart:async';
+
+import 'package:basic_auth_challenge/auth_app.dart';
+import 'package:basic_auth_challenge/core/di/app_injection.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_flavor/flutter_flavor.dart';
+
+import 'data/network/constants/network_constants.dart';
+
+void main() => initializeMain(
+      () => FlavorConfig(
+    name: kReleaseMode ? "Production" : "Development",
+    variables: {
+      'baseUrl': kReleaseMode ? NetworkConstants.BASE_URL_PRD : NetworkConstants.BASE_URL_DEV,
+      'connectionTimeout': NetworkConstants.CONNECTION_TIMEOUT,
+      'receiveTimeout': NetworkConstants.RECEIVE_TIMEOUT,
+      'isLive': kReleaseMode ? true : false,
+    },
+  ),
+);
+
+typedef FlavorCallback = FlavorConfig Function();
+final GlobalKey<NavigatorState> navigator = GlobalKey<NavigatorState>();
+
+Future<void> initializeMain(FlavorCallback flavorCallback) async {
+  await runZonedGuarded(() async {
```

Comment:
Good use of runZonedGuarded to capture all kind of errors. This is handy when it comes to CrashAnalytics